---
title: "Incorporate patient ID to ANOVA model"
author: "shelley herbrich"
date: "`r Sys.Date()`"
output: github_document
editor_options: 
  chunk_output_type: console
---

# Overview

# Librares and Data

## Libraries

```{r load_libs}

library(here)
library(RColorBrewer)
library(wesanderson) #currently requires devtools install 3/22/18
library(limma)

```

## Color Palette

```{r color_palette}
wes <- c(wes_palette("Darjeeling1"), wes_palette("Darjeeling2"))
```

## Data

```{r load_data}

load(here("results", "ng_data.RData"))

```

### Modifying analysis to include patient identifiers

Gene expression profiles for AML samples vary significantly between patients. We, therefore, would like to exploit the fact that we have multiple cell extracts from the same patient and use this information to extract genes that are differentially expressed be LSC+ and LSC- cells adjusting for source.

First, we confirmed that all samples collected from the same patient were run in the same batch. This is true for all but one sample, and that sample appears to be a re-run. Thus, depending on how we handle the data, the batch effect may be orthogonal (and thus completely removable) if we adjust for patient.


```{r describe_patients}
sample_annot$batch <- sapply(as.character(sample_annot$batch),function(x) strsplit(x,"_")[[1]][2])

## number of batches
## number of samples
table(apply(table(sample_annot$batch,sample_annot$patientID),2,function(x) sum(x!=0)))

patientsWithBoth <- names(which(sapply(split(sample_annot,factor(sample_annot$patientID)), function(x) length(unique(x[,"lsc_status"])))==2))
### total number of patients
length(unique(sample_annot$patientID))
### number of patients with samples that both did and did not engraft
length(patientsWithBoth)
datWB <- dat[,rownames(sample_annot)[which(sample_annot$patientID %in% patientsWithBoth)]]
mapWB <- sample_annot[rownames(sample_annot)[which(sample_annot$patientID %in% patientsWithBoth)],]
```

We use an ANOVA approach to look for genes which are significantly differentially expressed between engrafting and non-engrafting samples when we account for batch and patient source using the 48 of 83 patients that had cell fractions of both types. The summary difference in expression was determined as the average of the mean difference between LSC+ and LSC- cells from the same patient. We now identify 237 unique genes with p-values less than 0.01 (after Benjaminiâ€“Hochberg correction for multiple testing) and absolute fold-change greater than 2, as opposed to 104 genes in the original analysis . The most significantly different genes are listed below.

```{r aov_analysis}
aovDat <- t(apply(dat,1,function(x) summary(aov(x~factor(sample_annot$patientID)+factor(sample_annot$batch)+factor(sample_annot$lsc_status)))[[1]][,5]))
aovDat <- aovDat[,-4] #subtract empty column
colnames(aovDat) <- c("patient","batch","lsc")

# or if you want cell_fraction
#aovDat <- t(apply(dat,1,function(x) summary(aov(x~factor(sample_annot$patientID)+factor(sample_annot$batch)+factor(sample_annot$lsc_status)+factor(sample_annot$cell_fraction)))[[1]][,5]))
```

```{r calculate_diffs}
diffs <- t(apply(datWB,1, function(tt){
  df <- data.frame(exp=tt,LSC=mapWB$lsc_status)
  mean(sapply(split(df,factor(mapWB$patientID)), function(t) diff(sapply(split(t[,"exp"],t[,"LSC"]),mean))))
}))
names(diffs) <- rownames(datWB)
```


```{r multiple_test_corrections}
BHpvals <- p.adjust(aovDat[,"lsc"], method = "BH")
aovDat <- data.frame(aovDat,adjustedLSC=BHpvals,diff=t(diffs))

newProbes <- rownames(aovDat)[which(aovDat[,"adjustedLSC"] <= 0.01 & abs(aovDat[,"diff"]) >=1)]
newGeneList <- data.frame(geneID=unlist(probe_to_gene)[newProbes], aovDat[newProbes,c("adjustedLSC","diff")])  
newGeneList <- newGeneList[order(newGeneList$diff,decreasing = TRUE),]
length(unique(newGeneList$geneID))
head(newGeneList)
tail(newGeneList)

save(datWB, mapWB, aovDat, diffs, BHpvals, newGeneList, file= here("results","aov_analysis.RData"))
```
