---
title: "Compare original and modified analyses"
author: "shelley herbrich"
date: "`r Sys.Date()`"
output: github_document
editor_options: 
  chunk_output_type: console
---

# Overview

## Libraries

```{r load_libs}

library(here)
library(RColorBrewer)
library(wesanderson) #currently requires devtools install 3/22/18
library(limma)
library(ggplot2)
library(gridExtra)
library(grid)

```

## Color Palette

```{r color_palette}
wes <- c(wes_palette("Darjeeling1"), wes_palette("Darjeeling2"))
```

## Data

```{r load_data}

load(here("results", "ng_data.RData"))
load(here("results", "manuscript_results.RData"))
load(here("results", "aov_analysis.RData"))

```

### Comparing original and modified analyses

We compare the all fold changes and p-values obtained between the original analysis and the modified analysis that accounts for sample pairing in the plots below. The fold change values appears more consistent between analyses than p-value.

```{r plot_comparison}
df <- data.frame(reportedP= log(1/(manuscriptResults$adj.P.Val))*sign(manuscriptResults$logFC),patientAdjP= log(1/(aovDat[rownames(manuscriptResults),"lsc"]))*sign(diffs[rownames(manuscriptResults)]))

m <- ggplot(df, aes(x = reportedP, y = patientAdjP)) + geom_point(alpha = 1/10) 

hist_top <- ggplot(df)+geom_histogram(aes(reportedP),binwidth = 1) + theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank(), axis.title.y=element_blank())
empty <- ggplot()+geom_point(aes(1,1), colour="white")+
         theme(axis.ticks=element_blank(), 
               panel.background=element_blank(), 
               axis.text.x=element_blank(), axis.text.y=element_blank(),           
               axis.title.x=element_blank(), axis.title.y=element_blank())

scatter <- m + geom_vline(xintercept = c(-1,1)*log(1/0.5/nrow(manuscriptResults)), color="red") + geom_hline(yintercept = c(-1,1)*log(1/0.5/nrow(manuscriptResults)), color="red")
hist_right <- ggplot(df)+geom_histogram(aes(patientAdjP),binwidth = 1)+coord_flip() + theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank(), axis.title.y=element_blank(),axis.text.y=element_blank())


grid.arrange(hist_top, empty, scatter, hist_right, ncol=2, nrow=2, widths=c(4, 1), heights=c(1, 4))

####

df <- data.frame(reportedFC= manuscriptResults$logFC,patientAdjFC= diffs[rownames(manuscriptResults)])
m <- ggplot(df, aes(x = reportedFC, y = patientAdjFC)) + geom_point(alpha = 1/10) 

hist_top <- ggplot(df)+geom_histogram(aes(reportedFC),binwidth = .1) + theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank(), axis.title.y=element_blank())
empty <- ggplot()+geom_point(aes(1,1), colour="white")+
         theme(axis.ticks=element_blank(), 
               panel.background=element_blank(), 
               axis.text.x=element_blank(), axis.text.y=element_blank(),           
               axis.title.x=element_blank(), axis.title.y=element_blank())

scatter <- m + geom_vline(xintercept = c(-1,1), color="red") + geom_hline(yintercept = c(-1,1), color="red")
hist_right <- ggplot(df)+geom_histogram(aes(patientAdjFC),binwidth = .1)+coord_flip() + theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank(), axis.title.y=element_blank(),axis.text.y=element_blank())


grid.arrange(hist_top, empty, scatter, hist_right, ncol=2, nrow=2, widths=c(4, 1), heights=c(1, 4))
```

### Common genes

Of the original list of significantly differentially expressed genes, approximately 86% were captured in the modified analysis. The common up- and down-regulated genes from both analyses include (with selected examples):

```{r count_overlap}
originalUp <- actualList$geneID[which(actualList$logFC>0)]
originalDown <- actualList$geneID[which(actualList$logFC<0)]

modifiedUp <- as.character(unique(newGeneList$geneID[which(newGeneList$diff>=1)]))
modifiedDown <- as.character(unique(newGeneList$geneID[which(newGeneList$diff<= -1)]))
length(intersect(c(originalUp,originalDown), c(modifiedUp,modifiedDown)))/length(c(originalUp,originalDown))*100

## Common genes higher in LSC+ samples
sort(intersect(originalUp, modifiedUp))
length(intersect(originalUp, modifiedUp))/length(originalUp)*100

## Common genes higher in LSC- samples
sort(intersect(originalDown, modifiedDown))

length(intersect(originalDown, modifiedDown))/length(originalDown)*100
```


```{r inPatientPlot}
inPatientPlot <- function(i, probeN=1){
  all_probes <- which(unlist(probe_to_gene)==i)
  pi <- names(all_probes)[probeN]
  tt <- datWB[pi,]
  df <- data.frame(exp=as.numeric(tt),LSC=mapWB$lsc_status)
  tmp <- t(sapply(split(df,factor(mapWB$patientID)), function(t) sapply(split(t[,"exp"],t[,"LSC"]),mean)))
  direction <- rep("down",nrow(tmp))
  direction[which( apply(tmp,1,diff) >0)] <- "up"
  colTmp <- data.frame(sample = rep(rownames(tmp),2), direction=rep(direction,2), type= rep(c("LSC-","LSC+"),each=nrow(tmp)), exp= c(tmp))

  p <- ggplot(colTmp, aes(x = type, y = exp)) +
    geom_boxplot(outlier.shape = NA) +
    geom_line(aes(group = sample, color = factor(direction))) +
    geom_point() + labs(y=paste0(i," (log2 expression)")) + 
    theme(axis.title.x=element_blank()) + theme(legend.position = "top")
  
  allPatients <- data.frame(expression= dat[pi,], lsc=factor(sample_annot$lsc_status))

  q <- ggplot(allPatients, aes(x=expression, fill=lsc)) + geom_density(alpha=0.3) +
    theme(legend.position = "top") + labs(x=paste0(i," (log2 expression)"))
  
    q <- ggplot(allPatients, aes(x=expression, fill=lsc)) + geom_density(alpha=0.3) +
    theme(legend.position = "top",panel.background = element_rect(fill = "white",color="black"),panel.grid.minor = element_line(color="white")) + labs(x=paste0(i," (log2 expression)"))
 
    grid.arrange(q, p, ncol=2, nrow=1, widths=c(3, 3), heights=c(3), top=textGrob(i, gp=gpar(fontsize=15,font=8)))
}
```

```{r sigGenes}
inPatientPlot("CD34")
inPatientPlot("CD200")
```

###Genes gained with modified analysis

Significant genes we gain when accounting for sample pairing include (with selected examples):

```{r genes_gained}
sort(modifiedUp[which(!(modifiedUp %in% originalUp))])
sort(modifiedDown[which(!(modifiedDown %in% originalDown))])
inPatientPlot("MN1") 
inPatientPlot("EGFL7") 
```

###Genes lost with modified analysis

Significant genes that fall out include (with selected examples):

```{r genes_lost}
sort(originalUp[which(!(originalUp %in% modifiedUp))])
sort(originalDown[which(!(originalDown %in% modifiedDown))])
#inPatientPlot("TNFRSF4") 
#inPatientPlot("GPR56") 
```

## Examining the sources of variability

Using the results from the above ANOVA analysis, we can see which variables were most significant in explaining gene expression variability. By plotting the p-values corresponding to either ‘batch’, ‘LSC status’, and ‘sample ID’, we see that ‘sample ID’, by far, has the most extreme p-values. This confirms our hypothesis that the majority of differential gene expression is due to the sample source.


```{r plot_pvals}
sigs <- names(which(aovDat[,"adjustedLSC"] < 0.01))
sigDat <- dat[sigs,]
dim(sigDat)

aovDF <- data.frame(pval=(c(aovDat[,1],aovDat[,2],aovDat[,3])),source=rep(c("patient","batch","lsc"),each=nrow(aovDat)))
ggplot(aovDF, aes(source, pval,fill=source)) + geom_boxplot()  + coord_flip() +scale_fill_manual(values=c("#999999", "#E69F00", "#56B4E9")) + theme_bw()
```
